Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -189,6 +191,40 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Override if the algorithm is not part of the Mantid distribution.
   const std::string helpURL() const override { return ""; }
 
+  using IPropertyManager::declareProperty;
+
+  using PropertyManagerOwner::declareProperty;
+
+  void declareProperty(std::unique_ptr<Kernel::Property> p,
+                       const std::string &doc = "") override;
+
+  Kernel::IPropertyManager::TypedValue
+  getProperty(const std::string &name) const override;
```

Comment:
Yes, see `m_reservedList`. If a user attempts to access a property which is compound, an exception will be thrown. This is why it was necessary to override getProperty/declareProperty/setProperty.